docs/api/api_docs.yaml

```
@@ -976,27 +976,111 @@ paths:
+
+        If the `meta` query parameter is set then the response will be in JSON with the additional metadata tags.
+      parameters:
+        - $ref: '#/components/parameters/rev'
+        - schema:
+            type: boolean
+            default: 'false'
+          in: query
+          name: meta
+          description: Include the metadata of the attachment in the response body. Will cause `Content-Type=application/json` in the response header.
     put:
       responses:
-        '200':
-          description: OK
+        '201':
+          description: Attachment added to new or existing document successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/New-revision'
+          headers:
+            Etag:
+              schema:
+                type: string
+              description: The ID of the new revision.
+        '404':
+          $ref: '#/components/responses/Not-found'
+        '409':
+          $ref: '#/components/responses/Conflict'
       tags:
         - Admin
         - Public
+      summary: Create or update an attachment on a document
+      description: |-
+        This request adds or updates an attachment associated with the document. If the document does not exist, it will be created and the attachment will be added to it.
+
+        The maximum content size of an attachment is 20MB. The `Content-Type` header of the request should be what the content type of the attachment is otherwise this will default to `application/octet-stream`.
+
+        If the attachment already exists, the data of the existing attachment will be replaced in the new revision. The revisions are for tracking document changes, not attachment changes.
+
+        To remove an attachment from a document, simply update the `_attachments` dictionary of the document in the `PUT /{db}/{docid}` request.
+      parameters:
+        - name: Content-Type
+          schema:
+            type: string
+            default: application/octet-stream
+          in: header
+          description: The content type of the attachment.
+        - name: rev
+          schema:
+            type: string
+          in: query
+          description: The document revision to add/modify the attachment on. Required only when modifying an existing document. This falls back on the `If-Match` header if not set.
+        - name: If-Match
+          schema:
+            type: string
+          in: header
+          description: Used as the `rev` query parameter if it is not set.
```

Wording of this is ambiguous and exposes some unnecessary implementation detail.

```suggestion
          description: The existing document revision ID to modify. Required only when modifying an existing document.
        - name: If-Match
          schema:
            type: string
          in: header
          description: An alternative way of specifying the document revision ID.
```